### ISSUE TITLE
Wrong format string in logs

### ISSUE THREAD

**Issue by telenieko** [matrix-org/matrix-synapse-ldap3#51]
Hi, setting up authentication against an Active Directory LDAP I got this:

```
2018-09-05 16:10:36,908 - twisted - 131 - ERROR -  - Traceback (most recent call last):
2018-09-05 16:10:36,909 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/handlers.py", line 76, in emit
2018-09-05 16:10:36,910 - twisted - 131 - ERROR -  -     if self.shouldRollover(record):
2018-09-05 16:10:36,911 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/handlers.py", line 156, in shouldRollover
2018-09-05 16:10:36,912 - twisted - 131 - ERROR -  -     msg = "%s\n" % self.format(record)
2018-09-05 16:10:36,912 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 734, in format
2018-09-05 16:10:36,913 - twisted - 131 - ERROR -  -     return fmt.format(record)
2018-09-05 16:10:36,913 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 465, in format
2018-09-05 16:10:36,913 - twisted - 131 - ERROR -  -     record.message = record.getMessage()
2018-09-05 16:10:36,913 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 329, in getMessage
2018-09-05 16:10:36,913 - twisted - 131 - ERROR -  -     msg = msg % self.args
2018-09-05 16:10:36,914 - twisted - 131 - ERROR -  - TypeError: %d format: a number is required, not unicode
2018-09-05 16:10:36,914 - twisted - 131 - ERROR -  - Logged from file ldap_auth_provider.py, line 202
2018-09-05 16:10:36,914 - twisted - 131 - ERROR -  - Traceback (most recent call last):
2018-09-05 16:10:36,914 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 861, in emit
2018-09-05 16:10:36,914 - twisted - 131 - ERROR -  -     msg = self.format(record)
2018-09-05 16:10:36,914 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 734, in format
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  -     return fmt.format(record)
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 465, in format
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  -     record.message = record.getMessage()
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  -   File "/usr/lib/python2.7/logging/__init__.py", line 329, in getMessage
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  -     msg = msg % self.args
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  - TypeError: %d format: a number is required, not unicode
2018-09-05 16:10:36,915 - twisted - 131 - ERROR -  - Logged from file ldap_auth_provider.py, line 202
```

Which comes from this:
```python
                    logger.info(
                        "Registration based on LDAP data was successful: "
                        "%d: %s (%s, %)",
                        user_id, localpart, name, mail
                    )
```

But user_id appears to be a string, changing %d to %s works, and what is logged is a string.

**Comment by andrewshadura** [comment]
beae90f53 actually fixed this.

**Comment by richvdh** [comment]
yup fixed in #48
